core/src/test/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1672,6 +1696,25 @@ protected Settings nodeSettings(int nodeOrdinal) {
     }
 
     /**
+     * Returns a collection of plugins that should be loaded on each node.
+     */
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.emptyList();
+    }
+
+    /**
+     * Returns a collection of plugins that should be loaded when creating a transport client.
+     */
+    protected Collection<Class<? extends Plugin>> transportClientPlugins() {
+        return Collections.emptyList();
+    }
+
+    /** Helper method to create list of plugins without specifying generic types. */
+    protected static Collection<Class<? extends Plugin>> pluginList(Class<? extends Plugin>... plugins) {
```

Comment:
@robinst This is already the case on master (see https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java), which was done as part of a large warnings cleanup there. While that warnings cleanup is unlikely to be backported, would you like to open a PR to backport this fix?